### PR TITLE
Added support for SoundJS loading extentionless files and multiple paths

### PR DESCRIPTION
**Pull request by OJayRobinson**
Added required changes to support SoundJS loading extentionless files and multiple paths by setting src to an object that SoundJS parses and replaces with an appropriate src.
- added support for loading files with src set to an object that is then decoded by a plugin
- fix github issue 124, added error handler to TagRequest

### CODE REVIEW

**Comment by wdamien** [comment]
Can you remove the build files from the commit? I think they're causing merge conflicts.

**Comment by OJayRobinson** [comment]
Removed the build files and resolved the merge conflict

File: src/preloadjs/LoadQueue.js [hunk 1]
```diff
@@ -1337,22 +1337,14 @@ this.createjs = this.createjs || {};
...
-			item.type = createjs.RequestUtils.getTypeByExtension(item.ext);
-		}
-
-		// Inject path & basePath
 		var bp = ""; // Store the generated basePath
 		var useBasePath = basePath || this._basePath;
-		var autoId = item.src;
-		if (!match.absolute && !match.relative) {
+
+		if(item.src instanceof Object) {
```
Comment (wdamien) [inline src/preloadjs/LoadQueue.js]: Fix formatting ```if (```

File: src/preloadjs/LoadQueue.js [hunk 2]
```diff
@@ -1337,22 +1337,14 @@ this.createjs = this.createjs || {};
...
+			if(!item.type) {return null;} // the the src is an object, type is required to pass off to plugin
```
Comment (wdamien) [inline src/preloadjs/LoadQueue.js]: same

File: src/preloadjs/net/TagRequest.js [hunk 3]
```diff
@@ -141,6 +142,16 @@ this.createjs = this.createjs || {};
 	};
 
 	/**
+	 * Handle any error events from the tag.
+	 * @method _handleError
+	 * @protected
+	 */
+	p._handleError = function() {
+		this._clean();
+		this.dispatchEvent("error");
```
Comment (wdamien) [inline src/preloadjs/net/TagRequest.js]: Use existing this_sendError() method.

**Comment by wdamien** [comment]
The rest looks great. A supporting unit test would be awesome, but that could be another PR.

**Comment by OJayRobinson** [comment]
Pushed the if formatting changes.

TagRequest does not appear to have a _sendError method, internally or up the inheritance tree.

**Comment by wdamien** [comment]
Ah, my bad _sendError() is in the AbstractLoader not the AbstractRequest.

**Pull request merged**